### ISSUE TITLE
Add escaping to auto-generated version strings

### ISSUE THREAD

**Issue by 631068264** [scrapinghub/shub#213]
```
Packing version a71f746-feature/pic
Deploying to Scrapy Cloud project "112954"
Error: Deploy failed (400):
version: This value does not match the required pattern.
```

What is **required pattern** ?

**Comment by jdemaeyer** [comment]
Hey @631068264,
thanks for the report!

Version strings need to follow [Docker's limitations on tag names](https://docs.docker.com/engine/reference/commandline/tag/). In your case, it struggles with the forward slash in the branch name. For now, you can do one of the following (most preferable first):
1. [rename the branch](https://gist.github.com/lttlrck/9628955), or
2. specify a custom version string during deploy: `shub deploy --version feature_pic`, or
3. specify a custom version string in your `scrapinghub.yml`: `version: feature_pic` (on the top level)

In the long run, `shub` should properly escape forbidden characters from branch names when auto-generating version strings. At the very least, we should return a more helpful error message.

**Comment by jdemaeyer** [comment]
closed via #215
